src/sagemaker_xgboost_container/algorithm_mode/metrics.py

```diff
@@ -12,43 +12,15 @@
 # language governing permissions and limitations under the License.
 from sagemaker_algorithm_toolkit import metrics as m
 
+from sagemaker_xgboost_container.constants.xgb_constants import XGB_METRIC_NAMES
+
 
 # https://docs.aws.amazon.com/sagemaker/latest/dg/xgboost-tuning.html
 def initialize():
-    return m.Metrics(
-        m.Metric(name="validation:accuracy",
-                 direction=m.Metric.MAXIMIZE,
-                 regex=".*\\[[0-9]+\\].*#011validation-accuracy:(\\S+)"),
-        m.Metric(name="validation:auc",
-                 direction=m.Metric.MAXIMIZE,
-                 regex=".*\\[[0-9]+\\].*#011validation-auc:(\\S+)"),
-        m.Metric(name="validation:error",
-                 direction=m.Metric.MINIMIZE,
-                 regex=".*\\[[0-9]+\\].*#011validation-error:(\\S+)"),
-        m.Metric(name="validation:f1",
-                 direction=m.Metric.MAXIMIZE,
-                 regex=".*\\[[0-9]+\\].*#011validation-f1:(\\S+)"),
-        m.Metric(name="validation:logloss",
-                 direction=m.Metric.MINIMIZE,
-                 regex=".*\\[[0-9]+\\].*#011validation-logloss:(\\S+)"),
-        m.Metric(name="validation:mae",
-                 direction=m.Metric.MINIMIZE,
-                 regex=".*\\[[0-9]+\\].*#011validation-mae:(\\S+)"),
-        m.Metric(name="validation:map",
-                 direction=m.Metric.MAXIMIZE,
-                 regex=".*\\[[0-9]+\\].*#011validation-map:(\\S+)"),
-        m.Metric(name="validation:merror",
-                 direction=m.Metric.MINIMIZE,
-                 regex=".*\\[[0-9]+\\].*#011validation-merror:(\\S+)"),
-        m.Metric(name="validation:mlogloss",
-                 direction=m.Metric.MINIMIZE,
-                 regex=".*\\[[0-9]+\\].*#011validation-mlogloss:(\\S+)"),
-        m.Metric(name="validation:mse",
-                 direction=m.Metric.MINIMIZE,
-                 regex=".*\\[[0-9]+\\].*#011validation-mse:(\\S+)"),
-        m.Metric(name="validation:ndcg",
-                 direction=m.Metric.MAXIMIZE,
-                 regex=".*\\[[0-9]+\\].*#011validation-ndcg:(\\S+)"),
-        m.Metric(name="validation:rmse",
-                 direction=m.Metric.MINIMIZE,
-                 regex=".*\\[[0-9]+\\].*#011validation-rmse:(\\S+)"))
+
+    metrics = [m.Metric(name="validation:{}".format(metric_name),
+                        direction=m.Metric.MAXIMIZE,
```

Comment:
Not clear to me on how direction is being set based on metric name

Reply:
We might have to split into two lists or have a tuple for each entry with second item indicating the direction.

Reply:
I've added it back as it was before, with the addition of aucpr.